File: app/src/main/java/com/stormphoenix/ogit/adapters/GitOrgsAdapter.java
package com.stormphoenix.ogit.adapters;

import android.animation.Animator;
import android.animation.ObjectAnimator;
import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.stormphoenix.httpknife.github.GitOrganization;
import com.stormphoenix.ogit.R;
import com.stormphoenix.ogit.adapters.GitOrgsAdapter.OrgViewHolder;
import com.stormphoenix.ogit.adapters.base.BaseRecyclerAdapter;
import com.stormphoenix.ogit.utils.ImageUtils;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import de.hdodenhof.circleimageview.CircleImageView;

/**
 * Created by StormPhoenix on 17-3-11.
 * StormPhoenix is a intelligent Android developer.
 */

public class GitOrgsAdapter extends BaseRecyclerAdapter<GitOrganization, OrgViewHolder> {

    public GitOrgsAdapter(List<GitOrganization> dataList) {
        this(null, dataList);
    }

    public GitOrgsAdapter(Context context, List<GitOrganization> dataList) {
        super(context, dataList);
    }

    @Override
    protected Animator[] getAnimators(View itemView) {
        if (itemView.getMeasuredHeight() <= 0) {
            ObjectAnimator scaleX = ObjectAnimator.ofFloat(itemView, "scaleX", 1.05f, 1.0f);
            ObjectAnimator scaleY = ObjectAnimator.ofFloat(itemView, "scaleY", 1.05f, 1.0f);
            return new ObjectAnimator[]{scaleX, scaleY};
        }
        return new Animator[]{
                ObjectAnimator.ofFloat(itemView, "scaleX", 1.05f, 1.0f),
                ObjectAnimator.ofFloat(itemView, "scaleY", 1.05f, 1.0f),
        };
    }

    @Override
    public OrgViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.item_recyclerview_git_owner, parent, false);
        return new OrgViewHolder(view);
    }

    @Override
    public void onBindViewHolder(OrgViewHolder holder, int position) {
        super.onBindViewHolder(holder, position);
        OrgViewHolder viewHolder = (OrgViewHolder) holder;
        viewHolder.bind(data.get(position));
    }

    public static class OrgViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.header_image)
        CircleImageView mHeaderImage;
        @BindView(R.id.text_user_name)
        TextView mTextUserName;
        @BindView(R.id.text_other_info)
        TextView mTextOtherInfo;

        public OrgViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }

        public void bind(GitOrganization org) {
            ImageUtils.getInstance().displayImage(org.getAvatarUrl(), mHeaderImage);
            mTextUserName.setText(org.getLogin());
            if (!TextUtils.isEmpty(org.getDescription())) {
                mTextOtherInfo.setVisibility(View.VISIBLE);
                mTextOtherInfo.setText(org.getDescription());
            }
        }
    }
}

File: app/src/main/java/com/stormphoenix/ogit/adapters/GitOrgsAdapter.kt
package com.stormphoenix.ogit.adapters

import android.animation.Animator
import android.animation.ObjectAnimator
import android.content.Context
import android.support.v7.widget.RecyclerView
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import butterknife.BindView
import com.stormphoenix.httpknife.github.GitOrganization
import com.stormphoenix.ogit.R
import com.stormphoenix.ogit.adapters.GitOrgsAdapter.OrgViewHolder
import com.stormphoenix.ogit.adapters.base.BaseRecyclerAdapter
import com.stormphoenix.ogit.utils.ImageUtils
import de.hdodenhof.circleimageview.CircleImageView
import java.util.*

/**
 * Created by StormPhoenix on 17-3-11.
 * StormPhoenix is a intelligent Android developer.
 */

class GitOrgsAdapter(context: Context, dataList: MutableList<GitOrganization>) : BaseRecyclerAdapter<GitOrganization, OrgViewHolder>(context, dataList) {

    override fun getAnimators(itemView: View): List<Animator> {
        if (itemView.measuredHeight <= 0) {
            val scaleX = ObjectAnimator.ofFloat(itemView, "scaleX", 1.05f, 1.0f)
            val scaleY = ObjectAnimator.ofFloat(itemView, "scaleY", 1.05f, 1.0f)
            return Arrays.asList<Animator>(scaleX, scaleY)
        }
        return Arrays.asList<Animator>(
                ObjectAnimator.ofFloat(itemView, "scaleX", 1.05f, 1.0f),
                ObjectAnimator.ofFloat(itemView, "scaleY", 1.05f, 1.0f))
    }

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): OrgViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_recyclerview_git_owner, parent, false)
        return OrgViewHolder(view)
    }

    override fun onBindViewHolder(holder: OrgViewHolder, position: Int) {
        super.onBindViewHolder(holder, position)
        holder.bind(data[position])
    }

    class OrgViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var mHeaderImage: CircleImageView? = null
        internal var mTextUserName: TextView? = null
        internal var mTextOtherInfo: TextView? = null

        init {
            mHeaderImage = itemView.findViewById(R.id.header_image) as CircleImageView?
            mTextUserName = itemView.findViewById(R.id.text_user_name) as TextView?
            mTextOtherInfo = itemView.findViewById(R.id.text_other_info) as TextView?
        }

        fun bind(org: GitOrganization) {
            ImageUtils.getInstance().displayImage(org.avatarUrl, mHeaderImage)
            mTextUserName!!.text = org.login
            if (!TextUtils.isEmpty(org.description)) {
                mTextOtherInfo!!.visibility = View.VISIBLE
                mTextOtherInfo!!.text = org.description
            }
        }
    }
}
